### PR TITLE
updated the num_streams property on the engine class to get the acutal num_streams value being used

### PR DESCRIPTION
**Pull request by SageMoore**

### CODE REVIEW

**Comment by SageMoore** [comment]
We need to wait for the latest version of nyann to get pulled in, but this diff is ready for review.

File: src/deepsparse/engine.py [hunk 1]
```diff
@@ -284,7 +284,7 @@ def num_streams(self) -> int:
         :return: The max count of streams the current instance can handle
            concurrently.
         """
-        return self._num_streams
+        return self._eng_net.num_streams()
```
Comment (mgoin) [inline src/deepsparse/engine.py:287]: could we `self._num_streams = self._eng_net.num_streams()` after the `_eng_net` is constructed so there aren't two possible sources of truth?

Comment (bfineran) [reply src/deepsparse/engine.py:287]: if not already, can we add documentation on how the engine chooses the default `num_streams`?

Comment (SageMoore) [reply src/deepsparse/engine.py:287]: I'd prefer to just delete the instance variable all together. It's no longer used after construction.

**Review by tlrmchlsmth — APPROVED** [review]

**Review by mgoin — APPROVED** [review]
nice

**Pull request merged**